fw/rbcx-coprocessor/include/utils/Regulator.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <stdint.h>
+
+template <class value_type> struct Regulator {
+    typedef uint32_t coef_type;
+    typedef int32_t int_type;
+
+    constexpr Regulator<value_type>& operator=(Regulator<value_type>&& other)
+        = default;
+
+    Regulator() = default;
+    Regulator(const value_type& max_output, const coef_type& p,
+        const coef_type& i, const coef_type& d)
+        : m_w(0)
+        , m_stop(false)
+        , m_p(p)
+        , m_i(i)
+        , m_d(d)
+        , m_e(0)
+        , m_le(0)
+        , m_de(0)
+        , m_x(0)
+        , m_integrator(0)
+        , m_max_output(max_output) {}
+
+    void operator()(value_type v) { m_w = v; }
+    value_type operator()() { return m_w; }
```

Comment:
Uf, to se mi moc nelíbí. Proč to není pojmenovaná metoda, nebo se nepředává do `process()`?

Reply:
Mňo mě taky ne, jsem to vyžral z Aranky. Spravím to.